pkgs/development/python-modules/playwright/default.nix

```
@@ -0,0 +1,223 @@
+      outputHash = {
+        x86_64-darwin = "1nd6ll6dj0aqic3hh65y0br051ckbcap8spqbvzm19g8dn7vgv7m";
+      }.${system};
+    } ''
+      export PLAYWRIGHT_BROWSERS_PATH=$out
+      ${driver}/playwright.sh install
+      rm -r $out/.links
+    '';
+
+    dontConfigure = true;
+    dontBuild = true;
+
+    installPhase = ''
+      mkdir $out
+      cp -r * $out/
+    '';
+
+    doCheck = false;
+  };
+
+  browsers-linux = let
+    fontconfig = makeFontsConf {
+      fontDirectories = [];
+    };
+  in runCommand "playwright-browsers" {
+    nativeBuildInputs = [ makeWrapper ];
+  } ''
+    mkdir -p $out/chromium-1005/chrome-linux $out/ffmpeg-1007
+
+    # See here for the Chrome options:
+    # https://github.com/NixOS/nixpkgs/issues/136207#issuecomment-908637738
+    makeWrapper ${chromium}/bin/chromium $out/chromium-1005/chrome-linux/chrome \
+      --set SSL_CERT_FILE ${cacert}/etc/ssl/certs/ca-bundle.crt \
+      --set FONTCONFIG_FILE ${fontconfig}
+
+    ln -s ${ffmpeg}/bin/ffmpeg $out/ffmpeg-1007/ffmpeg-linux
+  '';
+in
+buildPythonPackage rec {
+  pname = "playwright";
+  version = "1.25.1";
+  disabled = pythonOlder "3.7";
+
+  src = fetchFromGitHub {
+    owner = "microsoft";
+    repo = "playwright-python";
+    rev = "v${version}";
+    sha256 = "sha256-TWc/fzRl5w4+kMF8odJfFwt+JSpCfyK4BJNB2mq8oTM=";
+  };
+
+  propagatedBuildInputs = [
+    autobahn
+    greenlet
+    objgraph
+    pillow
+    pixelmatch
+    pyee
+    pyopenssl
+    requests
+    service-identity
+    websockets
+  ];
+
+  format = "pyproject";
+
+  postPatch = ''
+    # The package seems to work with pyee 9 as well, which is what nixpkgs has packaged.
+    substituteInPlace meta.yaml --replace "- pyee ==8.1.0" ""
+    substituteInPlace setup.py \
+        --replace '"pyee==8.1.0",' "" \
+        --replace '"setuptools-scm==6.3.2",' "" \
+        --replace '"websockets==10.1",' ""
+
+    # Replace the functionality that setuptools-scm provides by adding the version to a file.
+    echo 'version = "${version}"' > playwright/_repo_version.py
+  '';
+
+  preBuild = ''
+    mkdir -p driver
+    cp ${driver.src} ./driver/${driver.filename}
+  '';
+
+  preInstall = ''
+    # pip names the wheel "dist", but when installing it will expect a properly formatted filename
+    # in dist.
+    mv dist dist.orig
+    mkdir -p dist
+    mv dist.orig dist/playwright-${version}-py3-none-any.whl
+  '';
+
+  postInstall = ''
+    pushd $out/lib/${python.libPrefix}/site-packages/playwright
+
+    # The setup script extracts the driver, but we do that ourselves, so it needs to be deleted
+    # again. See here:
+    # https://github.com/microsoft/playwright-python/blob/b5273b84d749a5064cc6e0329c900e1fdcb0e744/setup.py#L144-L162
+    rm -r driver
+    ln -s ${driver} driver
+
+    popd
+  '';
+
+  # Skip tests because they require network access.
+  doCheck = false;
+
+  pythonImportsCheck = [
+    "playwright"
+  ];
+
+  passthru = {
+    inherit driver browsers-mac browsers-linux;
```

```suggestion
    inherit driver browsers;
```
and browsers should be aliases to the right platform or does that not work?